$OpenBSD: patch-source3_wscript,v 1.10 2019/12/04 00:17:53 jca Exp $

- use more generic check for OpenBSD.
- avoid nested function error with clang

Index: source3/wscript
--- source3/wscript.orig
+++ source3/wscript
@@ -91,7 +91,7 @@ def configure(conf):
         conf.ADD_CFLAGS('-DDEVELOPER -DDEBUG_PASSWORD')
         conf.env.developer = True
 
-    if sys.platform != 'openbsd5':
+    if not sys.platform.startswith('openbsd'):
         conf.ADD_LDFLAGS("-Wl,--export-dynamic", testflags=True)
 
     # We crash without vfs_default
@@ -1522,6 +1522,7 @@ main() {
 
     conf.CHECK_CODE('void seekdir(DIR *d, long loc) { return; }',
                     'SEEKDIR_RETURNS_VOID',
+                    addmain=False,
                     headers='sys/types.h dirent.h',
                     msg='Checking whether seekdir returns void')
 
